### ISSUE TITLE
Only 1 audio track out of 2 available in playback

### ISSUE THREAD

**Issue by m-reiter** [xbmc/xbmc#16211]
### Describe the bug
Here is a clear and concise description of what the problem is:

I have a MythTV recording that offers 2 audio tracks: Original audio and a commented track for the visually impaired. I can confirm that both tracks are actually present in the recording as I can select between them when playing the .ts file through VLC.

However, playing the recording back in Kodi, only one stream is available. Selecting "change audio track" only presents one selection "ger" which, unfortunately, is the commented track.

I've experienced the issue in Kodi 18 both on Android (Fire TV stick) and on a Mac. I've also tried playing the file back from an NFS share instead of through the PVR client but got the same behaviour.

FFmpeg info for the file in question is:

```$ ffmpeg -i 10769_20190519181200.ts 
ffmpeg version N-93686-g8019395889-tessus  https://evermeet.cx/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
  built with Apple LLVM version 10.0.1 (clang-1001.0.46.4)
  configuration: --cc=/usr/bin/clang --prefix=/opt/ffmpeg --extra-version=tessus --enable-avisynth --enable-fontconfig --enable-gpl --enable-libaom --enable-libass --enable-libbluray --enable-libdav1d --enable-libfreetype --enable-libgsm --enable-libmodplug --enable-libmp3lame --enable-libmysofa --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-version3 --pkg-config-flags=--static --disable-ffplay
  libavutil      56. 26.100 / 56. 26.100
  libavcodec     58. 52.100 / 58. 52.100
  libavformat    58. 27.103 / 58. 27.103
  libavdevice    58.  7.100 / 58.  7.100
  libavfilter     7. 50.100 /  7. 50.100
  libswscale      5.  4.100 /  5.  4.100
  libswresample   3.  4.100 /  3.  4.100
  libpostproc    55.  4.100 / 55.  4.100
[mpegts @ 0x7fc689800000] PES packet size mismatch
    Last message repeated 1 times
[mpegts @ 0x7fc689800000] Could not find codec parameters for stream 3 (Unknown: none ([5][0][0][0] / 0x0005)): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[mpegts @ 0x7fc689800000] Could not find codec parameters for stream 4 (Unknown: none ([12][0][0][0] / 0x000C)): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[mpegts @ 0x7fc689800000] Could not find codec parameters for stream 5 (Unknown: none ([11][0][0][0] / 0x000B)): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[mpegts @ 0x7fc689800000] Could not find codec parameters for stream 6 (Unknown: none ([11][0][0][0] / 0x000B)): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Input #0, mpegts, from '10769_20190519181200.ts':
  Duration: 01:37:56.65, start: 89874.736200, bitrate: 3445 kb/s
  Program 1 
    Stream #0:0[0x65]: Video: hevc (Main) ([36][0][0][0] / 0x0024), yuv420p(tv), 1920x1080 [SAR 1:1 DAR 16:9], 50 fps, 50 tbr, 90k tbn, 50 tbc
    Stream #0:1[0x66](deu): Audio: aac_latm (LC) ([17][0][0][0] / 0x0011), 48000 Hz, stereo, fltp
    Stream #0:2[0x68](deu): Audio: aac_latm (LC) ([17][0][0][0] / 0x0011), 48000 Hz, stereo, fltp (visual impaired) (descriptions)
    Stream #0:3[0x492]: Unknown: none ([5][0][0][0] / 0x0005)
    Stream #0:4[0x498]: Unknown: none ([12][0][0][0] / 0x000C)
    Stream #0:5[0x1433]: Unknown: none ([11][0][0][0] / 0x000B)
    Stream #0:6[0x87c]: Unknown: none ([11][0][0][0] / 0x000B)
    Stream #0:7[0x6b](deu,deu): Subtitle: dvb_teletext ([6][0][0][0] / 0x0006), 492x250
```

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Upon playback, Kodi plays the audio from Stream #0:1. Using the audio playback menu I can choose between both audio tracks.

## Actual Behavior

Kodi plays the audio for the hearing impaired (Stream #0:2). In the audio playback menu, it says it's playing track 1/1 and the track selection is greyed out.

### To Reproduce

Play back the recording in question. Unfortunately, the file is 2.4 GB and certainly copyrighted. If anybody can suggest a method to extract a sample preserving all streams I'll happily do that. My own attempts at that have failed so far.

Otherwise, is there a possibility to get the file to someone (hopefully) looking into this without making it publicly available?

### Debuglog

The debuglog can be found here:

https://paste.kodi.tv/efuvelofof.kodi

Playback starts around line 3360.

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android on Fire TV stick 4K. From the log file: "Running on Amazon AFTMM with Android 7.1.2 API level 25, kernel: Linux ARM 32-bit version 4.4.1"
 - Kodi version: 18.1 Git:20190217-8cfdc89

**Comment by m-reiter** [comment]
After another look at the log file myself, I guess the problem shows in lines 3392ff:

```2019-05-28 12:24:20.750 T:10099    INFO: ffmpeg[7F9FF920]: Input #0, mpegts, from 'nfs://10.0.0.13/mnt/HD/HD_a2/Movies/10769_20190519181200.ts':
2019-05-28 12:24:20.751 T:10099    INFO: ffmpeg[7F9FF920]:   Duration: 01:37:56.65, start: 89874.736200, bitrate: 3445 kb/s
2019-05-28 12:24:20.751 T:10099    INFO: ffmpeg[7F9FF920]:   Program 1
2019-05-28 12:24:20.751 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:0[0x65]: Video: hevc (Main) ([36][0][0][0] / 0x0024), yuv420p(tv), 1920x1080 [SAR 1:1 DAR 16:9], 50 fps, 50 tbr, 90k tbn, 50 tbc
2019-05-28 12:24:20.751 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:1[0x66](deu): Unknown: none ([17][0][0][0] / 0x0011)
2019-05-28 12:24:20.752 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:2[0x68](deu): Audio: aac_latm (LC) ([17][0][0][0] / 0x0011), 48000 Hz, stereo, fltp (visual impaired)
2019-05-28 12:24:20.752 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:3[0x492]: Unknown: none ([5][0][0][0] / 0x0005)
2019-05-28 12:24:20.752 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:4[0x498]: Unknown: none ([12][0][0][0] / 0x000C)
2019-05-28 12:24:20.753 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:5[0x1433]: Unknown: none ([11][0][0][0] / 0x000B)
2019-05-28 12:24:20.753 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:6[0x87c]: Unknown: none ([11][0][0][0] / 0x000B)
2019-05-28 12:24:20.753 T:10099    INFO: ffmpeg[7F9FF920]:     Stream #0:7[0x6b](deu,deu): Subtitle: dvb_teletext ([6][0][0][0] / 0x0006)
```

Comparing that to the ffmpeg output I posted above, the built-in(?) ffmpeg in Kodi seems to have trouble decoding/recognising the audio stream in question.

To verify that, is there an easy way to get the Mac build of Kodi to use an external ffmpeg binary? I guess I'm out of look on the non-rooted FireTV stick in this regard, right?

**Comment by m-reiter** [comment]
I don't expect anybody to solve this for me but I'd really appreciate if somebody could give me a hint on how to look into this further myself.

**Comment by m-reiter** [comment]
Update: This is still present in the 18.3 release (both on the FireTV stick and on Mac) but seems to be fixed in the latest 18.3 nightlies.

Given the lack of interest this created, I doubt anybody will want to look into what exactly fixed it, so I'm closing the issue.

**Comment by a1rwulf** [comment]
@m-reiter can you give us access to your sample file?

**Comment by m-reiter** [comment]
To quote myself: 

> Unfortunately, the file is 2.4 GB and certainly copyrighted. If anybody can suggest a method to extract a sample preserving all streams I'll happily do that. My own attempts at that have failed so far.
> 
> Otherwise, is there a possibility to get the file to someone (hopefully) looking into this without making it publicly available?

So any suggestions on the best method to provide a/the sample file to you?

**Comment by m-reiter** [comment]
Just had an idea myself. I'll try to record a short excerpt from a different programme and see whether that shows the same behaviour. In that case I'll provide that.

**Comment by a1rwulf** [comment]
>  ffmpeg -ss 00:01:00 -i input.ts -to 00:04:00 -c copy output.ts

Something like this should work to cut out a small sample.

**Comment by m-reiter** [comment]
Just tried that but it only copied one audio track.

I have scheduled a couple of short recordings for later today in the hope that at least one of them will have the issue. If none does, we can have another look at extracting a sample from the existing recordings.

**Comment by FernetMenta** [comment]
> what exactly fixed it

skipping avformat_find_stream_info completely for mpegts fixed it:
https://github.com/xbmc/xbmc/pull/16314

**Comment by a1rwulf** [comment]
Thx @FernetMenta for making the cross reference.
@m-reiter At this point I think we can close it.
I see there's a backport PR merged as well, so if we decide to do 18.4 it will be in there.

**Comment by m-reiter** [comment]
I agree that this can be closed now. I'm happy for now and it's good to know this will make it into the next release.

@a1rwulf @FernetMenta Thanks for taking an interest!
